attest/core/src/quote.rs

```diff
@@ -165,76 +166,77 @@ impl Quote {
         self.0.len()
     }
 
+    fn try_get_slice(&self, range: Range<usize>) -> Result<&[u8], EncodingError> {
+        if self.capacity() < range.end {
+            Err(EncodingError::InvalidInputLength)
+        } else {
+            Ok(&self.0[range])
+        }
+    }
+
     /// Read the quote version
-    pub fn version(&self) -> u16 {
-        u16::from_le_bytes(
-            (&self.0[QUOTE_VERSION_START..QUOTE_VERSION_END])
-                .try_into()
-                .unwrap(),
-        )
+    pub fn version(&self) -> Result<u16, EncodingError> {
```

Comment:
The alternative to this sort of change is to manually implement the `serde` traits to prevent reading improper values.  That will be the approach taken in the new-style crates, but I'd rather not fiddle with serde serialization, since this object will be deprecated, first by debaidufication, then by DCAP.